File: Java/2024. Maximize the Confusion of an Exam.java
class Solution {
    public int maxConsecutiveAnswers(String answerKey, int k) {
        int l = 0, t = 0, f = 0, ans = 0;
        for (int r = 0; r < answerKey.length(); r++) {
            char c = answerKey.charAt(r);
            if (c == 'T') {
                t++;
            } else {
                f++;
            }

            while (Math.min(t, f) > k) {
                if (answerKey.charAt(l) == 'T') {
                    t--;
                } else {
                    f--;
                }
                l++;
            }

            ans = Math.max(ans, r - l + 1);
        }
        return ans;
    }
}

File: Python3/2024. Maximize the Confusion of an Exam.py
class Solution:
    def maxConsecutiveAnswers(self, answerKey: str, k: int) -> int:
        ans = k
        count = Counter(answerKey[:k])
        l = 0

        for r, c in enumerate(answerKey[k:], k):
            count[c] += 1

            while min(count['T'], count['F']) > k:
                count[answerKey[l]] -= 1
                l += 1

            ans = max(ans, r - l + 1)

        return ans